quark/radiocontrast.py

```diff
@@ -0,0 +1,137 @@
+        :param depth: maximum number of recursive search functions.
+        :return: a set of first_method_list ∩ second_method_list or None.
+        """
+
+        # Not found same method usage, try to find the next layer.
+        depth += 1
+        if depth > self.max_search_layer:
+            return
+
+        # Append first layer into next layer.
+        next_level_set = method_set.copy()
+
+        # Extend the xref from function into next layer.
+        for md in next_level_set:
+            if md[0].is_android_api():
+                self.api_set.add(md[0])
+                continue
+
+            self.method_recursive_search(self.apkinfo.lowerfunc(md[0]), depth)
+
+    def rule_generate(self):
+        """
+        Generate rules and export them to the output directory.
+
+        :return: None
+        """
+        # Rescursive search for apis in target method.
+        lower_funcs = set(self.apkinfo.lowerfunc(self.method))
+        self.method_recursive_search(lower_funcs)
+
+        first_apis_pool = list(self.api_set)
+        second_apis_pool = list(self.api_set)
+
+        # Setup progress bar
+        second_api_pool_num = len(second_apis_pool)
+        outter_loop = tqdm(first_apis_pool)
+
+        for api1 in first_apis_pool:
+            outter_loop.update(1)
+
+            for num, api2 in enumerate(second_apis_pool, start=1):
+                inner_desc = f"{num}/{second_api_pool_num}"
+                outter_loop.set_postfix(inner_loop=inner_desc, refresh=True)
+
+                # Skip the case of same method.
+                if api2.name == api1.name:
+                    continue
+
+                generated_rule = {
+                    "crime": "",
+                    "permission": [],
+                    "api": [
+                        {
+                            "class": api1.class_name,
+                            "method": api1.name,
+                            "descriptor": api1.descriptor,
+                        },
+                        {
+                            "class": api2.class_name,
+                            "method": api2.name,
+                            "descriptor": api2.descriptor,
+                        },
+                    ],
+                    "score": 1,
+                    "label": [],
+                }
+                comb = RuleObject("test", json_data=generated_rule)
+
+                try:
+                    self.quark.run(comb)
+                except KeyboardInterrupt:
+                    raise
+                except Exception as e:
+                    tqdm.write(
+                        "{} and {} combination has some error when analyzing,\
+                        ERROR MESSAGE: {}".format(
+                            api1, api2, e,
+                        ),
+                    )
+                    continue
+
+                if comb.check_item[4]:
+                    rule_name = f"{outter_loop.n}{num}.json"
```

Comment:
It seems like the filenames of the generated rules are random numbers, could you name them sequentially? For example, `0.json`, `1.json`, `2.json`, and so on.